django/db/models/sql/compiler.py

```diff
@@ -1811,23 +1811,54 @@ def as_sql(self):
             on_conflict=self.query.on_conflict,
         )
         result = ["%s %s" % (insert_statement, qn(opts.db_table))]
-        fields = self.query.fields or [opts.pk]
-        result.append("(%s)" % ", ".join(qn(f.column) for f in fields))
 
-        if self.query.fields:
-            value_rows = [
-                [
-                    self.prepare_value(field, self.pre_save_val(field, obj))
-                    for field in fields
+        if fields := self.query.fields:
+            from django.db.models.expressions import DatabaseDefault
+
+            supports_default_keyword_in_bulk_insert = (
+                self.connection.features.supports_default_keyword_in_bulk_insert
+            )
+            value_cols = []
+            for field in list(fields):
+                field_prepare = partial(self.prepare_value, field)
+                field_pre_save = partial(self.pre_save_val, field)
+                field_values = [
+                    field_prepare(field_pre_save(obj)) for obj in self.query.objs
                 ]
-                for obj in self.query.objs
-            ]
+                if field.has_db_default():
+                    # If all values are DEFAULT don't include the field and its
+                    # values in the query as they are redundant and could prevent
+                    # optimizations. This cannot be done if we're dealing with the
+                    # last field as INSERT statements require at least one.
+                    if len(fields) > 1 and all(
+                        isinstance(value, DatabaseDefault) for value in field_values
+                    ):
+                        fields.remove(field)
+                        continue
+                    elif not supports_default_keyword_in_bulk_insert:
+                        # If the field cannot be excluded from the INSERT for the
+                        # reasons listed above and the backend doesn't support the
+                        # DEFAULT keyword each values must be expanded into their
+                        # underlying expressions.
+                        prepared_db_default = field_prepare(field.db_default)
+                        field_values = [
+                            (
+                                prepared_db_default
```

Comment:
```suggestion
                        field_values = [
                            (
                                field_prepare(field.db_default)
```

Reply:
This is done explicitly. `field.db_default` is the same for all instances and `field_prepare` is not cheap so it's better to prepare it once and use the same value than prepare it N times.

Reply:
Ah, gotcha!